site/content/docs/browser-checks/scraping-onpage-elements.md

```diff
@@ -1,154 +1,142 @@
+})
  ```
 {{< /tab >}}
 {{< /tabs >}}
+
 Ok, let's break this down:
 
-1. We require all needed modules and load a page.
-2. We use `$page.eval()` and pass it two arguments: 1) a query selector string 2) a callback to handle the resulting element.
-3. We inspect the element's `textContent` property and use the `trim()` method to get rid of any trailing white space.
-4. We validate that indeed the name is as expected.
+1. We require `@playwright/test` library, declare a test block and go to the Checkly docs page.
+2. We use `page.getByRole()` and pass it two arguments:
+  - A role of the element
+  - An object with `name` property - it will narrow the search to an element with a matching text, or in this case, `aria-label` attribute 
+3. We expect the `searchBox` element to contain `CTRL-K` text (a keyboard shortcut info)
 
-> Some may want to use the `innerText` property here, but `textContent` is [actually better](https://stackoverflow.com/questions/35213147/difference-between-textcontent-vs-innertext)
 
 ## Scraping a list of values
 
-Using the `page.$$()` (notice the two dollar signs) we can scrape an array of elements in one go. You commonly use this
-when the query selector you are providing targets multiple similar elements on a page, say a list of links:
+Playwright Test makes it easy to work with [lists of elements](https://playwright.dev/docs/locators#lists). The `getByX()` methods will return a list, if multiple elements match the parameters. You could then assert the count, text or perform additional filtering on the elements:
 
-{{< tabs "page.$$eval" >}}
-{{< tab "Playwright" >}}
-```js
-const playwright = require('playwright')
-const expect = require('expect')
+{{< tabs "Scraping list values" >}}
+{{< tab "TypeScript" >}}
+```ts
+import { expect, test } from '@playwright/test'
 
-const browser = await playwright.chromium.launch()
-const page = await browser.newPage()
+test('Find "Advanced" sections on playwright docs page', async ({ page }) => {
+  await page.goto('https://playwright.dev/docs/intro')
 
-await page.goto('https://news.ycombinator.com/news')
-const stories = await page.$$eval('a.storylink', links =>
-  links.map(link => link.textContent).slice(0, 10)
-)
-expect(stories).toHaveLength(10)
-expect(typeof stories[0]).toBe('string')
+  const advancedSections = page.locator('.menu__link').filter({ hasText: /Advanced.+/i })
 
-await browser.close()
+  await expect(advancedSections).toHaveCount(2)
+  await expect(advancedSections).toHaveText(['Advanced: configuration', 'Advanced: fixtures'])
+})
 ```
 {{< /tab >}}
-{{< tab "Puppeteer" >}}
+{{< tab "JavaScript" >}}
 ```js
-const puppeteer = require('puppeteer')
-const expect = require('expect')
+const { expect, test } = require('@playwright/test')
 
-const browser = await puppeteer.launch()
-const page = await browser.newPage()
+test('Find "Advanced" sections on playwright docs page', async ({ page }) => {
+  await page.goto('https://playwright.dev/docs/intro')
```

Comment:
please add `// 1` etc. comments to make it easier to understand the explanations